### PR TITLE
Rise Cache 2.0 integration

### PR DESCRIPTION
**Pull request by ezequielc**
@alex-deaconu @settinghead Migration of RiseCacheController to JS and updating `ping()` and `getCacheVideoURL()` to support Rise Cache v2. Please review. Thanks!

### CODE REVIEW

File: src-js/Viewer/Cache/RiseCacheController.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  if (xhr.status === 200) {
+    isActive = true;
+    clearTimeout(pingInterval);
+
+  } else if (xhr.status === 404) {
+    Logger.log("Rise Cache v2 is running");
+    clearTimeout(pingInterval);
+    _isV2Running = true;
+
+    service.pingCache();
```
Comment (alex-deaconu) [inline src-js/Viewer/Cache/RiseCacheController.js]: @ezequielc curious, why `pingCache` again? Doesn't the `404` response already indicate that v2 is running? Thanks.

**Comment by settinghead** [comment]
@ezequielc :+1:

File: src-js/Viewer/Cache/RiseCacheController.js (hunk 1 above)
Comment (ezequielc) [inline src-js/Viewer/Cache/RiseCacheController.js]: @alex-deaconu I followed widget implementation: https://github.com/Rise-Vision/widget-common/blob/master/src/js/rise-cache.js#L29-L52 
Now that you mentioned I don't see a strong reason for the new request. Unless RCv1 (or another server in the same port) could eventually raise a 404 and we want to double check in the correct RCv2 url. Thoughts? Thanks

Comment (alex-deaconu) [inline src-js/Viewer/Cache/RiseCacheController.js]: @ezequielc got it. I'll leave it up to you if you want to remove the second request. Thanks.

**Comment by alex-deaconu** [comment]
@ezequielc code looks good 👍 . I did not verify the behaviour with RC1 or RC2, I'll leave that up to you. Thanks!

File: src-js/Viewer/Cache/RiseCacheController.js (hunk 1 above)
Comment (ezequielc) [inline src-js/Viewer/Cache/RiseCacheController.js]: @alex-deaconu As there might be a reason we don't know, I prefer to leave the second request as is. Thanks!

**Pull request merged**